### PR TITLE
Added ObservationApi based off PatientApi and added it to OpenApi interface

### PR DESCRIPTION
**Pull request by that-guy-is-it**

### CODE REVIEW

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/OpenApi.java [hunk 1]
```diff
@@ -17,4 +18,4 @@
             description = "US Core Implementation Guide",
             url = "https://build.fhir.org/ig/HL7/US-Core-R4/index.html"))
 @Path("/")
-public interface OpenApi extends PatientApi {}
+public interface OpenApi extends PatientApi, ObservationApi {}
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/OpenApi.java:21]: alphabetical

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              required = true,
+              description =
+                  "The logical id of the resource." + " Once assigned, this value never changes.")
+          String id);
+
+  @Operation(
+      summary = "Observation Update",
+      description = "https://www.hl7.org/fhir/r4/observation.html",
+      tags = {"Observation"})
+  @GET
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java:58]: PUT

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+              mediaType = "application/fhir+json",
+              schema = @Schema(implementation = OperationOutcome.class)))
+  @ApiResponse(
+      responseCode = "404",
+      description = "Not found",
+      content =
+          @Content(
+              mediaType = "application/fhir+json",
+              schema = @Schema(implementation = OperationOutcome.class)))
+  ResponseEntity<Void> observationUpdate(
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java:81]: `ResponseEntity<Void>` is a Spring-specific implementation detail. For the purposes of generating swagger, it's just `void`

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      @Parameter(
+              in = ParameterIn.PATH,
+              name = "id",
+              required = true,
+              description =
+                  "The logical id of the resource." + "Once assigned, this value never changes.")
+          String id,
+      @Valid @RequestBody Observation observation);
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java:89]: Remove these two annotations and use the Swagger `@RequestBody` annotation instead

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+              schema = @Schema(implementation = ResponseEntity.class)))
+  @ApiResponse(
+      responseCode = "400",
+      description = "Bad request",
+      content =
+          @Content(
+              mediaType = "application/fhir+json",
+              schema = @Schema(implementation = OperationOutcome.class)))
+  @ApiResponse(
+      responseCode = "404",
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java:75]: You can't get a `not-found` while saving

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+  @Operation(
+      summary = "Observation Update",
+      description = "https://www.hl7.org/fhir/r4/observation.html",
+      tags = {"Observation"})
+  @GET
+  @Path("Observation/{id}")
+  @ApiResponse(
+      responseCode = "200",
+      description = "Record found",
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java:62]: 200 is returned for update; 201 is returned for creating a new record

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+              schema = @Schema(implementation = OperationOutcome.class)))
+  void observationUpdate(
+      @Parameter(
+              in = ParameterIn.PATH,
+              name = "id",
+              required = true,
+              description =
+                  "The logical id of the resource." + " Once assigned, this value never changes.")
+          String id,
+      @RequestBody Observation observation);
```
Comment (apeirats-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationApi.java:82]: `@RequestBody` should be tagged as required.

**Review by apeirats-va — APPROVED** [review]

**Pull request merged**